lib/event.c

```diff
@@ -721,14 +833,31 @@ void event_master_free(struct event_loop *m)
 	pthread_cond_destroy(&m->cancel_cond);
 	close(m->io_pipe[0]);
 	close(m->io_pipe[1]);
+#if !(defined(DISABLE_EPOLL)) &&                                               \
+	(defined(HAVE_EPOLL_WAIT) || defined(HAVE_EPOLL_PWAIT) ||              \
+	 defined(HAVE_EPOLL_PWAIT2))
+	epoll_ctl(m->handler.epoll_fd, EPOLL_CTL_DEL, m->io_pipe[0], NULL);
+	close(m->handler.epoll_fd);
+#endif
 	list_delete(&m->cancel_req);
 	m->cancel_req = NULL;
 
 	hash_clean_and_free(&m->cpu_record, cpu_record_hash_free);
 
 	XFREE(MTYPE_EVENT_MASTER, m->name);
+#if !(defined(DISABLE_EPOLL)) &&                                               \
```

Comment:
that is ok. But I guess CI system uses default compling settings, and USE_EPOLL will not be flagged. In that way the code tested by CI will be upstream code (using poll) but not modified code (using epoll). Seems not good. How to do with that?

Reply:
@eqvinox

Reply:
@eqvinox I have replaced DISABLE_EPOLL with USE_EPOLL and that should work fine. Now users can enable event loop with epoll APIs with *./configure CFLAGS="-DUSE_EPOLL" ...* provided. Otherwise, poll APIs in upstream codes are used.

The problem is that after executing git rebase yesterday, I can't pass CI test. This is strange since codes in current commit are pre-compiled without USE_EPOLL, so output executable programs should be executing workflow of current upstream/master commit, which should be bug-free and supposed to pass tests. I found following tests often fail:
- Topotest part 7: test_static_routes_topo1_ebgp/test_same_rte_from_bgp_static_p0_tc5_ebgp
- Topotest part 9: ospf_sr_te_topo1/test_ospf_sr_te_topo1.py

Does this indicate current upstream/master is buggy or tests above are buggy?

Reply:
Finally passed anyway. But I feel like there are some problem with CI. Probably because tests consumes lots of computing resource, causing tests slow and expiring convergence time when the CI system is overloaded?